### ISSUE TITLE
Save redirect on "published" mode does not work as intended

### ISSUE THREAD

**Issue by fcisio** [marcelofinamorvieira/datocms-plugin-slug-redirects#2]
Hey, sorry to bring this up again.
I think the new mode doesn't behave as expected.

The array seems to be appended in reverse or something.

### current

```json
{
  {
    "source": "first-of-them-all-01", // should be "first-of-them-all"
    "destination": "first-of-them-all", // should be "first-of-them-all-01"
    "urlPrefix": "/posts/",
    "recordID": "133731803"
  },
  {
    "source": "third-post-01", // should be "third-post"
    "destination": "third-post", // should be "third-post-01"
    "urlPrefix": "/posts/",
    "recordID": "135975908"
  },
  {
    "source": "third-post-02", // right value
    "destination": "third-post-01", // should be "third-post-03"
    "urlPrefix": "/posts/",
    "recordID": "135975908"
  }
}
```

**Comment by marcelofinamorvieira** [comment]
Hello @fcisio 

I wasn't able to reproduce the issue, could you share with us a screen recording of the steps you took to get such a order?
I tried a similar test and i got the expected result:

```
[
  {
    "source": "zero",
    "destination": "one",
    "urlPrefix": "",
    "recordID": "54830185"
  },
  {
    "source": "one",
    "destination": "two",
    "urlPrefix": "",
    "recordID": "54830185"
  },
  {
    "source": "two",
    "destination": "three",
    "urlPrefix": "",
    "recordID": "54830185"
  },
  {
    "source": "three",
    "destination": "four ",
    "urlPrefix": "",
    "recordID": "54830185"
  }
]
```

**Comment by fcisio** [comment]
Sure, here's a [screen recording](https://drive.google.com/file/d/19DpcTLLIgL2HSeCP2y2GZBNjzgBV9krF/view?usp=share_link).

Note that this issue is specific to the "Add redirect rule only upon record publication" mode enabled.
Using the default "save" mode works as intended.

**Comment by fcisio** [comment]
Hi, is there any progress on this?
Were you able to reproduce the issue?

**Comment by fcisio** [comment]
I also noticed that a redirect is created even is the slug value hasn't changed.

**Comment by marcelofinamorvieira** [comment]
@fcisio sorry for the delay on this, just released v0.5.0 that fixes these issues :)

**Comment by fcisio** [comment]
Hey, thanks for this update! It seems to work as intended on my end.
The only thing left is that it creates redirects even when the slug value hasn't changed.

```json
{
    "source": "welcome-to-styllar-universe",
    "destination": "welcome-to-styllar-universe",
    "urlPrefix": "/posts/",
    "recordID": "154992715"
  },
```

I believe a simple check like this should do the trick:

```js
const oldSlug = recordBeforeUpdate[updatedFieldKey];
const newSlug = (createOrUpdateItemPayload.data.attributes as object)[
  updatedFieldKey
];

if (oldSlug !== newSlug) {
  updateSlugRedirects(
    urlPrefix,
    oldSlug as string,
    newSlug,
    recordBeforeUpdate.id,
    client
  );
}

return true
```
